tools/stats/upload_external_contrib_stats.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import datetime
+import json
+import os
+import urllib.parse
+from typing import Any, Callable, cast, Dict, List, Optional
+from urllib.error import HTTPError
+from urllib.request import Request, urlopen
+
+from tools.stats.upload_stats_lib import upload_to_s3
+
+FILTER_OUT_USERS = set(["pytorchmergebot", "facebook-github-bot", "pytorch-bot[bot]"])
+
+
+def _fetch_url(
+    url: str,
+    *,
+    headers: Optional[Dict[str, str]] = None,
```

Comment:
Would this work?  Then line 23-24 can be removed I think

```suggestion
    headers: Optional[Dict[str, str]] = {},
```